includes/models/model.llms.quiz.attempt.php

```diff
@@ -436,18 +438,51 @@ private function get_new_questions() {
 					$locks[] = $index;
 				}
 
-				$questions[] = array(
-					'id'      => $question->get( 'id' ),
-					'earned'  => 0,
-					'points'  => $question->supports( 'points' ) ? $question->get( 'points' ) : 0,
-					'answer'  => null,
-					'correct' => null,
+				/**
+				 * Filter each question to add extra parameters for the questions array.
+				 *
+				 * @since [version]
+				 *
+				 * @param array         $question The question array to be used for the quiz.
+				 * @param LLMS_Question $question The question object.
+				 */
+				$questions[] = apply_filters(
+					'llms_question_array',
+					array(
+						'id'      => $question->get( 'id' ),
+						'earned'  => 0,
+						'points'  => $question->supports( 'points' ) ? $question->get( 'points' ) : 0,
+						'answer'  => null,
+						'correct' => null,
+					),
+					$question
 				);
 
 			}
 
-			if ( $randomize ) {
-
+			/**
+			 * Filter questions for the quiz.
+			 *
+			 * Sets the questions to be used for the quiz.
+			 *
+			 * @since [version]
+			 *
+			 * @param array     $questions The questions to be used for the quiz.
+			 * @param LLMS_Quiz $llms_quiz LLMS_Quiz instance.
+			 */
+			$questions = apply_filters( 'llms_quiz_questions', $questions, $quiz );
+
+			/**
+			 * Filter randomize value for quiz questions.
+			 *
+			 * Sets the randomize to boolean value based on `Randomize` and `Question Bank`.
+			 *
+			 * @since [version]
+			 *
+			 * @param boolean   $randomize The randomize boolean value.
+			 * @param LLMS_Quiz $llms_quiz LLMS_Quiz instance.
```

Comment:
don't need the prefix in the param name, right? $quiz is enough.

Reply:
yup, fixed.